z3c/dependencychecker/package.py

```diff
@@ -21,32 +22,32 @@ class PackageMetadata(object):
 
     def __init__(self, path):
         logger.debug("Reading package metadata for %s...", path)
-        self.distribution_root = self._get_distribution_root(path)
-        self.setup_py_path = self._get_setup_py_path()
-        self.package_dir = self._get_package_dir()
-        self.egg_info_dir = self._get_egg_info_dir()
-        self.name = self._get_package_name()
+        self._distribution_root = None
+        self._path = path
+
         self._working_set = self._generate_working_set_with_ourselves()
-        self.top_level = self._get_sources_top_level()
 
-    @staticmethod
-    def _get_distribution_root(path):
-        if 'setup.py' not in os.listdir(path):
+    @cached_property
+    def distribution_root(self):
+        if 'setup.py' not in os.listdir(self._path):
             logger.error(
                 'setup.py was not found in %s. '
                 'Without it dependencychecker can not work.',
-                path,
+                self._path,
             )
             sys.exit(1)
-        return path
+        self._distribution_root = self._path
```

Comment:
Same remark as for "name" previously. Setting _distribution_root is a side effect.

Reply:
Right, rather than setting a ``self._distribution_root`` we can simply return ``self._path`` to avoid the side effect.

And as we cache the property, it does not matter that we are doing some checks there.